blocklist_de-feed/operations.py

```diff
@@ -0,0 +1,118 @@
+""" Copyright start
+  Copyright (C) 2008 - 2022 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+from connectors.core.connector import get_logger, ConnectorError
+import requests
+from time import sleep
+import arrow
+
+try:
+    from integrations.crudhub import trigger_ingest_playbook
+except:
+    # ignore. lower FSR version
+    pass
+
+logger = get_logger('blocklist_de-feed')
+
+SERVICE_MAPPING = {
+    "All": "all",
+    "SSH": 'ssh',
+    "Mail": 'mail',
+    "Apache": 'apache',
+    "IMAP": 'imap',
+    "FTP": 'ftp',
+    "SIP": 'sip',
+    "VOIP": "voip",  # not added
+    "Bots": 'bots',
+    "IRC Bot": "ircbot",
+    "Strong IPs": 'strongips',
+    "Brute Force Login": 'bruteforcelogin'
+}
+MAX_ATTEMPTS = 3
+
+
+def validate_response(response):
+    if response.ok:
+        result = response.text
+        if 'text/plain' in response.headers["Content-Type"] and 'error' not in result.lower():
+            return result
+    logger.exception('Fail To request API {0} response is : {1}'.
+                     format(str(response.url), str(response.content)))
+    raise ConnectorError(
+        'Fail To request API {0} response is : {1}'.format(str(response.url), str(response.content)))
+
+
+def make_request(config, url, parameters=None, method='GET'):
+    verify_ssl = config.get('verify_ssl')
+    attempt = 1
+    status_code = 404
+    api_response = None
+    while attempt <= MAX_ATTEMPTS:
+        try:
+            api_response = requests.request(method=method, url=url, params=parameters, verify=verify_ssl, timeout=90)
+            if api_response.ok or api_response.status_code == status_code:
+                return api_response
+        except Exception as e:
+            logger.exception(e)
+            if 'read timed out' in str(e).lower():
+                attempt += 1
+                sleep(1)
+                continue
+            raise ConnectorError(e)
+        attempt += 1
+        sleep(1)
+    return api_response
+
+
+def convert_to_unixtime(last_added_time):
+    try:
+        date_time = arrow.get(last_added_time).strftime("%Y-%m-%dT%H:%M:%S")
+        return date_time
+    except Exception as Err:
+        logger.exception('{0}'.format(str(Err)))
+        raise ConnectorError('{0}'.format(str(Err)))
+
+
+def fetch_indicators(config, params=None, **kwargs):
+    try:
+
+        service = params.get('service', 'strongips')
+        output_mode = params.get('output_mode')
+        last_added_time = params.get('time', None)
+        create_pb_id = params.get("create_pb_id")
+        query_param = {}
+        if last_added_time:
+            new_time = convert_to_unixtime(last_added_time)
+            url = 'https://api.blocklist.de/getlast.php?time={last_added_time}'.format(
```

Comment:
shouldn't the URL be read from config?

Reply:
There are 2 URLs to call API.
Pass time with URL: https://api.blocklist.de/
   Eg. https://api.blocklist.de/getlast.php?time=2022-02-11T00:00&service=ftp
Pass service with URL: https://lists.blocklist.de/. (Taken this URL as config param)
   Eg. https://lists.blocklist.de/ftp.txt

How can I take 2 URLs from config?